src/coreComponents/mesh/generators/PrismUtilities.hpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2020-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#ifndef GEOSX_MESH_GENERATORS_PRISMUTILITIES_HPP_
+#define GEOSX_MESH_GENERATORS_PRISMUTILITIES_HPP_
+
+#include "common/DataTypes.hpp"
+#include "common/Span.hpp"
+
+namespace geosx
+{
+
+/**
+ * @brief Get the local indices of the nodes in a face of the prism with N-sided polygon base.
+ * @tparam N the number of sides in the polygon base
+ * @param[in] faceNum the local index of the target face in the element  (this will be 0-numFacesInElement)
+ * @param[in] elemNodes Element to nodes mapping.
+ * @param[out] faceNodes space to write node indices to, must be of sufficient size
+ * @return number of nodes in the face
+ */
+template< localIndex N >
+localIndex getFaceNodesPrism( localIndex const faceNum,
```

Comment:
This is a strange numbering to get faces `1` and `last` as the polygonal faces. I guess it's the VTK numbering?

Reply:
I agree that it looks strange. I decided to follow the logic used for tetrahedra, pyramids, wedges and hexahedra, that assumes:

- lexicographic ordering
- face node ordering following the counter-clockwise rule (pointing outward)

Specifically:

```
Tetrahedron faces: 013
                   021
                   032
                   123

Pyramid faces: 014
               0231
               042
               134
               243

Wedge faces: 0154
             0231
             042
             135
             2453

Hexahedron faces: 0154
                  0231
                  0462
                  1375
                  2673
                  4576
```

Applying the same logic to a pentagonal prism gives:

```
Prism5 faces: 0165
              04321
              0594
              1276
              2387
              3498
              56789
```

I'm happy to change it.

Reply:
I didn't see the lexicographic ordering. I am fine with the way it is.

Reply:
@jeannepellerin Do you have any point of view w.r.t. these orderings?

Reply:
I propose we address this aspect in a follow-up PR.